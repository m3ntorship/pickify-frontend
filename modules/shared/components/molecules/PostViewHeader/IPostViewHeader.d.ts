declare namespace IPostViewHeader {
  export interface IProps {
    deletePostHandler: (postId: string) => void;
    userId: string;
    postId: string;
    profilePic?: string;
    isHidden: boolean;
    name?: string;
    date: string;
  }
  // The IData equals to IImgWithInfo.IData
}

export { IPostViewHeader };
