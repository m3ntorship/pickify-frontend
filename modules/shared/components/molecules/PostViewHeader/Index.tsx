import React from 'react';
import Divider from '../../atoms/Divider/Divider';
import type { ReactElement, FC } from 'react';
import VerticalThreeDotsIcon from '../../icons/verticalThreeDots.svg';
import type { PostViewHeaderTypes } from './PostViewHeaderTypes';
import UserInfo from '../UserInfo/Index';
import * as EDivider from '../../atoms/Divider/types/EDivider';

const PostViewHeader: FC<PostViewHeaderTypes.IProps> = ({
  size = 'medium',
  variant,
  imgSrc,
  name,
  date,
  handleEditIconClick,
  id,
}): ReactElement => {
  return (
    <div>
      <div className="flex justify-between items-start pb-s">
        <UserInfo
          size={size}
          variant={variant}
          imgSrc={imgSrc}
          name={name}
          date={date}
        />
        <VerticalThreeDotsIcon
          onClick={handleEditIconClick}
          className="fill-grey cursor-pointer"
          data-testid={id}
        />
      </div>
      <Divider type={EDivider.DividerType.Horizontal} length="100%" />
    </div>
  );
};

export default PostViewHeader;
