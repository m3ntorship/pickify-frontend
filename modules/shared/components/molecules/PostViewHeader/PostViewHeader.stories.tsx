import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import PostViewHeader from './PostViewHeader';
import type { IPostViewHeader } from './IPostViewHeader';

export default {
  title: 'molecules/PostViewHeader',
  component: PostViewHeader,
};

const Template: Story<IPostViewHeader.IProps> = (args): ReactElement => (
  <div className="flex justify-between">
    <PostViewHeader {...args} />
  </div>
);
export const postViewHeader = Template.bind({});
postViewHeader.args = {
  name: 'Marzouk el akta3',
  isHidden: false,
  date: '1967-11-01T19:15:45.339Z',
  profilePic:
    'data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAoHCBYWFRgVFRUYGBgZGBgYGBoYGBgYGBoYGhgZGhoYGBgcIS4lHB4rHxgYJjgmKy8xNTU1GiQ7QDs0Py40NTEBDAwMEA8QHhISHjQkJCsxNDQ0NDQ0NDQ0NDQ0NDQ0NDQ0NDQ0NDQ0NDQ0NDQ0NDQ0NDQ0MTQ0NDQ0NDQ0NDQ/NP/AABEIAOQA3QMBIgACEQEDEQH/xAAbAAABBQEBAAAAAAAAAAAAAAAEAAEDBQYCB//EADQQAAEDAwMCBAQGAQUBAAAAAAEAAhEDBCEFEjFBUSJhcZEGgaGxEzJCwdHwIxRScuHxFf/EABkBAAMBAQEAAAAAAAAAAAAAAAACAwEEBf/EACIRAAMAAgMAAgIDAAAAAAAAAAABAhEhAxIxQVETMiJCYf/aAAwDAQACEQMRAD8A9ECYpJKJYSQSSCAOk6ZILAHThMnC0BwnTBOtMEnBTJLQOwnBXATytMwdykuQV1KDB0k0pIA6lKUySAHTrlJAHSSaUpTAOkkkgBJJJIMAUkklIoJJMnQB0kCmlIlYaOXJByHuakDPv+xVG3W9ri09Dg+XZJXIp9HmHXhpWPBXayzdYh8g+vmtHbXTXiWkLY5FRl8dSTJJJBUJjpJJIASeUySYw6lPK4SQYSSkuHPA5KFF80mAcLHSXoylsNlJcMfK7laLgdJMnlACSSSTAJPKZJAAaSZOpDjJ0ySAHCZyZ5Qle4LeySqSHmWyK/eQDHt0Kwd5dAvI4Mq/1rVIHHtKwl7dS4unK5KfZnZxz1Wyw/1ZLoBWl0qq7HI9FldHol75gc9lsKLI8PTqsawbTyi6p6g4CJlO/UX+Sr2QAmcm719k+k/QaNbc3kAj2KJZr1OMyD2hZm6d2QLXmY81q5qQfhlmxfrw/S33XB12OWhZy3aXfZSvSvnv7D8Ml6dfH+36oavrrzxhUrx7qB9bpP1WPmp/Jq4pXwWjr17vzOx65K5ZfQVR1a56FQ/6o8hGW9j9UbajrUYKuLa6DhMheZ0arnGJj7rU6XXDQMqsczT2R5OJY0a5rl1KCtq4IRbXLsmkzkqcHaYuSJUL3JxSbem3KDcluQBGkkkpDiSKcoetcBqymktjSmzqqcKnvasTlPdah5qlv7okLk5KTOmJaKXWriZH8/ZZ5tPc8N5JPr9Ap7+qS8ycBH/DltJLyOfy/wAolYRWmaDT7YMZwJRf40BRhp+Sgrh3DQUlGSStuiSdvRS0bmcHkIa2ouAOOi6taBBL3deAlUvJraFXf+33Q7GeL7Luocp2MMgf3lDNXgVbtPyU4p44z/codr8/JNcXUD5T90JGMe4YB1EqsuWgZn6f2V1VvJBIx5lBsq78fXojBqFWeHDlVjakO2k/UouuwtVbfN/UOfunn6MYfSudp/KD8ytBYXsxKyFKruHEEK60+rjlZU4DOUba0r9irm2qysTbXe0q9s7rsVbjvBz8kZNGXKBxXNCvIyncV2J5WjlawMklKcIMGLlE+u0KO5qgBZ6/vT0K5r5Ovh0xx9izu9VAwAqW61HuYVNcXbp5KgY6SC7OcDzUKqq9OiYmS1Y4u8R46BCancBrT09Ua0Yysf8AEt0XP2A4BSTOawN8ZBqtUOIA/UeTyc8x2W1022hrRxAyspoltvqtn9Ik+XYLcOc1rOyq9LCEb2C6hfim0kAY78epWf8A/v1HEEl+2YBAa0Z7bsnhXV5Yh7GOePAXguHeQ8tB8vCFn9cfRBG38ffLGvGwbC135tm2CI42kme6txcSe2S5eXrpGv026loJzjOIz5+66fWBkBVPw08toQ/k/lnGBI+4Rop7jjr1UOXVYRXj/lOWclgny4H7pUx4seQRRowIAEBcCj1+igymiN4xPkhn1QRnz90TcgAEKnrtPIPqPJMgwPc7IAdgTn0Crz8RU2naxg2jAMwftH1S1Zm5rAD+YwSTAHm49AqKqA15aCw+MsOyDBp+EkEydrpnmCR5ALq4+JVOzn5eXrWEa5lw2o3qPXkfNV11RjByo7f/ABAxJY5oc3y7j0wi21A9uVGp6vRaa7IpGOh0DB6K20y4kw4c8qsvG7Xg9Dhc0rra8TwUzXZGLTNW54bHMfRGW1yWnmQq4t3swq23vHMdtPHT+FNLJrPRLC8lXNN0hYbS70T28lsLB4IXTw0/Dm5YDNi6DFI1dQuk5jIavfFp2gSVn7mueXH5K11thDpjlZi/qEyAY7nsvMrLo9OEuoPdXkGBk9AETYSCC4+Lt2VTSd4oYMn9R+6s7Vnb5nuU3iB7Ln8WeOyyOo0T+I5xWkpOzHv6Kr1tmCe6xPDNS0R/C9WS8x1HsAtPbne6DwMnzPZZT4Qt3Pc8DutnQo/hjy695VK9JLwJu27mOYeHCMciDII8wVUnTA8jed2cmNoPyV2x4c0BJw2hDujOsv1AH4AcQAIa3AAEY7QrC3oBvRK2Z5IgM6uwlY2RqrMYCBewo1x9fZROaISOafwOtFfUZuHRU17bQcSr2rTPIQN7TJCxBkpGUgQ5rpMiGntkH9lWt0WHk4OfQ+/ZXVHJg8qWoQOQqzdToWomvQdlEBh3xJEbRkR0z7n5quqN2GAcHhWT6jY4QFw3cEOsgkl4V+oOls9QhKomD7/sUfc0jtygqbZiEyejGto1WjvlkFVGr0i1+4d8qy0kbWGUFqh6H3ST6NQRpt1uAzkLVaTqrmwOR2XndrWLHfNaK1uZ9ehWvMvKMaVLDPU7O6DxKKWe+HHOLAStAu3jfZZOG1h4Mvr1HwkrC33UfRehfEI/xFYK5gNkef06rj5Vijs4X/Eq4DMfqPPkOwVjbHGBAVQxpc/1KuKA6JGVJ2GM90Nq7fD8lMDLgOybVRNP5LAO/gdgDHu7uK1NTIMLKfBlbwPaBw/75wtOKkKjeyJDSBwIg/dFVckN5XVNobk8qOgfESsQBtOBjt91O1oHJyoGYErJfGOrPY2WP2Z5x9FfjhJZD1mwrPAQ1SrAVFp2oPfbMe+Q5zcziSMTHmq74s1V9Kn4DkkCeY81RygVGgqOBMjlDVagj15HYqj0G7e+mHucCT1VjVfOVz8sLGR0wS8Ba6R9FHWqy2fdEXHib6IGk/lp46qIwMXlTsZgKR9uBnkLk1OiAIb6n4CVS2Akx5/ur+4/KfMdVS6O3xkdinn9WK/UXr3bWR1VbcncCi9UqQflCCYcLUtACAT9irXSmHeG+aqwIdC0Xw5TmqwHuj3QN4R6Vo1tsYI7KzAUVuyGgKVdsrCwefTyyl1pm6k4eRXn94zwfRekXbJYQsVfUPCRC5Of1M7OF6wZa2bnAVnTbCjFPbwFM1qhnJ0YE1q61Fn+L3XQau7tu5noUIxlF8NXWyq9hwDBlbeiZiF5fcvLK4Pp91vNJu920dwArUvGRXyaCoBEqFnGPJdlv/nH1U4ZjhLg3JxUJDcjHZVL9IoPO943EHh0uj0lXTxhUV68tJ2q8VgERatVEbWYaBAAQup0GPG1wkIO5rkB24eikdUcSTtkZ5x9FRvIaQHb6cKQ8D3NByWg49irGiYb1I80OxjnO3OgDiOyLEQocjyMsEbHSfIqF1LP8dR6IgMzhR16ZwRz1CgbkkpskYKCrsyUzrzaYP8Af5S1Gr4Wu74wtwwyD3D+iD0cf5T/AMiuH1z1U2iszu9SnS0xWzrVH+M+q5olRXL9z5UlHsg1HNZmZWk+FxNZioHsJWl+FQBVYShPaMr9WensGE6ZhwnXceeBPbKzuq2sGQMOWjUNeiHDKjc9kWiurMFe20FCNBV7qtqQTjCqxTXE5wztmsoZjUi3wPJ4yV0G5CD1q62M2Dk8/wALZnLMpmP1ITUnstN8NkyxxMZ5PYcwOpn+8LOMpFzvMn91eWD4dE43Bjf+LfLzwVevCaRuxUAcGzE+/v3RTnKh1ivsLHickRHcwAP72VjSug8DOeD6pQxrJOSYKratOXZ7x5IyrcBvosvrPxAyn3nsqSsmeBNcMaQTEHI8ueZ7ED5LkPDhMg+mZOAvPdU1qpVP5iBnj+9lJpetvpjbMtjA9v2VerwT/Ksm8Igc/wDiiLyDhV1DV2v65xI+UotrwSue0Wn7CmOKmMbZPuhA+Mzwgr7UchreIM/Lkfv79kszk0HvmbzuZmDmOfmPlyotSfDGM6jJ8p6KbSGxuce+PfCEvzuJdGTyPLoQt+QAa78eqs7Y7aZPUqkqO8YHmrVzsBvSE1LCQqeWcM5RdJqgpsRLFNlEFUmStF8NWwdVGJhUNs1b/wCF9O2M3O5K3jntQnLWJNFTGF2kEl3HACJFMnCQoQXFs14gjCo7vQiSSwwPNaNPCRxNejTbnwyDdGqAEmB2Ky2q2J3EEyZXqdZuCsTrtPxYUqlT4Wi3T2ZFlttcPUKe0bte1zuA8/P8qsnWoAnl3TsP5Q/+kyXPkgGf4CUoaC9Y17QOYIcPkoqrPwy8k5MRHHAgQhrOoS7PoETrs7A4D9In5YT8b9QJZaRFqNJ5pMe1wcXNJI7GSB68FYXUtOqOcXPH8Lf2uKDAR4nDecdHZA+qrb1o910zMoSk3pnnzrI5xwu2aa7GFpK1BsO7kQPmVZ3Nu0GI6BPjRLpsyNGze0yMFXthvdAMCSPbM/RFvtUXpsMducMCSe3BU6SZRJpaKP4gllRjWuO0tJdnrIifqnp0pJJwA0AevA+koC6eaj4GZJiejZMD6qzMNaGdhn1UaaSKTOFs7ZU2w3tkoC6uA9+OACobi6MmEKw8qan5MbI7kZlWlhXa8ATDuI7oEtkIVzYMjBCfCpYEy0zRlhCnp0yTACrbO6c4AOMrTaO/xCQCouXnBTssZLj4f0Qlwe8Y5hbykwAYQOmNG0YhWQC6+OVKOTkp09iTpQkqEgJOE8JkhQSdILoLAI3jCyuvWgbLicrXELL/ABLTdCnyfqU4vTK2ryXxKtbikCMccqoY2HR7qzpVQ4Z+SjLL0t5BTgzCtaTRVpkH0+RQFxSUltX/AAsn8vWE0PFGplhXAjHQQPIBU1y391bVniJHWCPQiQgKrQSV2D9SiuaeAO7m/cK3qUwfZC1afjYOm5v3RlVsLUxeuzgtBwoarBtc0/7T9ipmKG5rNYC5xgAZKShuuEVNtZNYN5mec90FdVee55UtzqYfO0ENHE4nz9FWvfOVz1tiusnDl2GcJjClY1AgtsIJzZKKuXqOiwlGcAlkmtsELUaa8YKztOkeyt7R5aFOmOpR6boV817QOqvNy810K9LXjOCt9SrSAV08VdpOPlnrQZuS3IXelvVSRH+Im/EQX4iW8qZXAaHp/wARA711uQGAz8RVOts3MRQK4q09whZSysBLwzC1aEEpWsh2VdanbAKoc2CuPDlnYn2RaOpg56BcGnIIhd27pbA6JbO6r/pPzRyKcU2x0EH06INzuqNpPDZbz2855CBuY6cEn5eo6FXiso6IeVgjEOePIz7CVI6SfkudPp7nkDt90TVZsdnpx6KvwZ/bBFTpc+6rNSpte1wJwAceaKvLrG1vJ+ndV9YQ2OVDktLRtaWCgrMjAUQRtemULsypJkWKmyVO4QFLTo4UNc8haZ6BlslWNtREIe3Z5SrFgAEpaY0rByfDhEU34QNxcjcI4RLaoIwkpDLYbbVSDK9C0G73sHcLzMPIWz+EbocFPwVisE+aczk2UJQkCnXacJWfhpBi6hOplhtqcNTpwgUcNTkJBOgCl1CjMqjr0wtRfNwqF9PxLmtbOmK0LT1JcnK4p+AFxQzK+92cD6onzAV7k5qDtypH0w8eLB6kdfUdUW1jQMBQ1AP7ymWgVNbRXUrZ7H7mPBz2j7FK4L3wXOMf9o1gAyhbioJgdk3avMj/AJX6Bse2S0eh9E1alyubRniPqi67IgqbRnYoq7Poo22s5R1xR8c9Cu3sgQtSMbBDSgKurNkq4eCW+YVfUZlDYSiO1Z5KxIAYZUVvSyn1QQ3HVKtsZ+FA/DiJ6o63dgIQDcfNHWzYW14E6DmDCs9MuCxwIKpi1GWhKl5sfGT0vSNQD2jKs5Xn+lXZY8ZwVubesHNBXbxX2Rw8sYYOnTSktMHThMkgDsLoLhdIAFvG4VXUpjlW9wJCrareila2Vh6KzVR4FTUjBWhv2eFUn4clSa2VT0FU7gkRwpYULcYAXRctMI69U8BClvdTuGZSczErDSC0ZnjkyrG6pSFzYUZMqe5Phd7KiWibeyje2SnLJXD6kOjurG3pTBQaA1aMfNAVLbKv7mn4UBswT2SMeWCNG3KrtUrzEKxvnQB5iQs++SShI3I9MAqztg2OEBRYi2YS0x0HtDTwpWCOAoqJEcKQuCQ0mY9X+m6rDIJ4WZD/ABKck9E005YjlUtnoSSSS7DjEnlOkgDmU8p0kARVDhBFJJSr0rHgNdDCrSwSkkkfpRHLkPUeZTJLDTl3RTv4CZJBgezDcKCv+X5JJJxEUF8ME9QcKz0p5LMpJINYUeCg2/q9EkkjHRQai87fSYVSxJJb8GoNtkexgTJKbGJHiAISBSSWAO/oi6fCSSGaj//Z',
};
