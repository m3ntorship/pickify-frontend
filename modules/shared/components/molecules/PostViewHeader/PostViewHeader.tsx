import React from 'react';
import type { ReactElement, FC } from 'react';
import { useSession } from 'next-auth/client';
import Divider from '../../atoms/Divider/Divider';
import VerticalThreeDotsIcon from '../../icons/verticalThreeDots.svg';
import type { IPostViewHeader } from './IPostViewHeader';
import * as EDivider from '../../atoms/Divider/types/EDivider';
import UserInfo from '../UserInfo/UserInfo';
import DropDown from '../../atoms/DropDown/DropDown';
import { options } from '../../atoms/DropDown/mockedOptions';

const getPostMenuOptions = (
  updatedOptions: { id: string; body: string }[],
  userId: string,
  uuid: string,
): { id: string; body: string }[] => {
  if (process.browser) {
    if (userId !== uuid) {
      return updatedOptions.filter((option) => option.id !== 'delete');
    }
    return updatedOptions;
  }
  return updatedOptions;
};

const PostViewHeader: FC<IPostViewHeader.IProps> = ({
  profilePic,
  name,
  userId,
  date,
  deletePostHandler,
  postId,
  isHidden,
}): ReactElement => {
  const [session] = useSession();

  const onMenuOptionClickHandler = (id: string): void => {
    switch (id) {
      case 'delete':
        deletePostHandler(postId);
        break;
      case 'report':
        console.log('reported');
        break;
      case 'save':
        console.log('saved');
        break;
      default:
        console.log('default');
    }
  };
  return (
    <div>
      <div className="flex justify-between items-start pb-s">
        <UserInfo
          isHidden={isHidden}
          profilePic={profilePic}
          name={name}
          date={date}
        />
        <DropDown
          onOptionMenuClick={onMenuOptionClickHandler}
          options={getPostMenuOptions(options, userId, session?.uuid as string)}
          variant="post"
          size="sm"
        >
          <VerticalThreeDotsIcon className="fill-grey w-6 h-6" />
        </DropDown>
      </div>
      <Divider type={EDivider.DividerType.Horizontal} length="100%" />
    </div>
  );
};

export default PostViewHeader;
