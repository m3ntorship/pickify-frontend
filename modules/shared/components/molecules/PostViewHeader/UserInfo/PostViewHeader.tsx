import React from 'react';
import type { ReactElement, FC } from 'react';
import Divider from '../../../atoms/Divider/Divider';
import VerticalThreeDotsIcon from '../../../icons/verticalThreeDots.svg';
import type { IPostViewHeader } from './IPostViewHeader';
import * as EDivider from '../../../atoms/Divider/types/EDivider';
import UserInfo from '../../UserInfo/UserInfo';

const PostViewHeader: FC<IPostViewHeader.IProps> = ({
  profilePic,
  name,
  date,
  handleEditIconClick,
  id,
  isHidden,
}): ReactElement => {
  return (
    <div>
      <div className="flex justify-between items-start pb-s">
        <UserInfo
          isHidden={isHidden}
          profilePic={profilePic}
          name={name}
          date={date}
        />
        <VerticalThreeDotsIcon
          onClick={handleEditIconClick}
          className="fill-grey cursor-pointer"
          data-testid={id}
        />
      </div>
      <Divider type={EDivider.DividerType.Horizontal} length="100%" />
    </div>
  );
};

export default PostViewHeader;
