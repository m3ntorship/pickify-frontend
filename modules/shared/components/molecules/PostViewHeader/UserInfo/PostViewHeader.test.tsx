import * as React from 'react';
import { render, screen } from '@testing-library/react';
import * as renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import type { TargetElement } from '@testing-library/user-event';
import PostViewHeader from './PostViewHeader';

describe('Testing postViewHeader with snapshot', () => {
  it('if isHidden is true, name becomes Anonymous as well even if a name is given', () => {
    const tree = renderer
      .create(
        <PostViewHeader
          date={new Date('1967-11-01T19:15:45.339Z')}
          id="test"
          isHidden
          name="Marzouk el akta3"
          handlePostOptionsIconClick={(): boolean => {
            return true;
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('Testing postViewHeader react testing library', () => {
  it('should be clicked once', () => {
    const handleClick = jest.fn();
    render(
      <PostViewHeader
        id="test"
        isHidden
        date={new Date('1967-11-01T19:15:45.339Z')}
        name="Marzouk el akta3"
        handlePostOptionsIconClick={handleClick}
      />,
    );
    const postOptionsIcon: TargetElement = screen.getByTestId('test');
    const timesOfClick = 1;
    userEvent.click(postOptionsIcon);
    expect(handleClick).toBeCalledTimes(timesOfClick);
  });
});
