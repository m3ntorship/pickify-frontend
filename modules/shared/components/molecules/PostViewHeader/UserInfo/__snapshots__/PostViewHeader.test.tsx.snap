// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing postViewHeader with snapshot if isHidden is true, name becomes Anonymous as well even if a name is given 1`] = `
<div>
  <div
    className="flex justify-between items-start pb-s"
  >
    <div
      className="outer-wrapper"
    >
      <div
        className="wrapper medium"
      >
        <svg
          className="absolute w-full h-full"
          data-file-name="SvgAnonymous"
          height={40}
          width={40}
        />
      </div>
      <div
        className="user-wrapper"
      >
        <span
          className="name"
          data-testid="name"
        >
          Anonymous
        </span>
        <span
          className="date"
          title="01/11/1967 21:15:45"
        >
          54 years ago
        </span>
      </div>
    </div>
    <svg
      className="fill-grey cursor-pointer"
      data-file-name="SvgVerticalThreeDots"
      data-testid="test"
      onClick={[Function]}
    />
  </div>
  <div
    className="bg-grey-shd6 h-px"
    style={
      Object {
        "width": "100%",
      }
    }
  />
</div>
`;
